iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/dataset/DataciteMDStoreImporter.java

```diff
@@ -79,66 +75,35 @@
 	@Override
 	public void run(PortBindings portBindings, Configuration conf,
 			Map<String, String> parameters) throws Exception {
-		FileSystem fs = FileSystem.get(conf);
-		if (!parameters.containsKey(IMPORT_DATACITE_MDSTORE_SERVICE_LOCATION)) {
-			throw new InvalidParameterException("unknown MDStore service location, "
-					+ "required parameter '" + IMPORT_DATACITE_MDSTORE_SERVICE_LOCATION + "' is missing!");
-		}
-		if (!parameters.containsKey(IMPORT_DATACITE_MDSTORE_IDS_CSV)) {
-			throw new InvalidParameterException("unknown MDStore identifier, "
-					+ "required parameter '" + IMPORT_DATACITE_MDSTORE_IDS_CSV + "' is missing!");
-		}
 		
-//		setting result set client read timeout
-		Long rsClientReadTimeout = null;
-		if (parameters.containsKey(IMPORT_RESULT_SET_CLIENT_READ_TIMEOUT)) {
-			rsClientReadTimeout = Long.valueOf(
-					parameters.get(IMPORT_RESULT_SET_CLIENT_READ_TIMEOUT));
-		}
+		Preconditions.checkArgument(parameters.containsKey(IMPORT_DATACITE_MDSTORE_IDS_CSV), 
+                "unspecified MDStore identifiers, required parameter '%s' is missing!", IMPORT_DATACITE_MDSTORE_IDS_CSV);
 		
-		DataFileWriter<DataSetReference> datasetRefWriter = null;
-		DataFileWriter<DocumentToMDStore> datasetToMDStoreWriter = null;
-		try {
-//			initializing avro datastore writers
-			datasetRefWriter = DataStore.create(
-					new FileSystemPath(fs, portBindings.getOutput().get(PORT_OUT_DATASET)), 
-					DataSetReference.SCHEMA$);
-			datasetToMDStoreWriter = DataStore.create(
-					new FileSystemPath(fs, portBindings.getOutput().get(PORT_OUT_DATASET_TO_MDSTORE)), 
-					DocumentToMDStore.SCHEMA$);
-			
+		FileSystem fs = FileSystem.get(conf);
+		
+        try (DataFileWriter<DataSetReference> datasetRefWriter = DataStore.create(
+                new FileSystemPath(fs, portBindings.getOutput().get(PORT_OUT_DATASET)), DataSetReference.SCHEMA$);
+                DataFileWriter<DocumentToMDStore> datasetToMDStoreWriter = DataStore.create(
+                        new FileSystemPath(fs, portBindings.getOutput().get(PORT_OUT_DATASET_TO_MDSTORE)),
+                        DocumentToMDStore.SCHEMA$)) {
```

Comment:
This initialize try resources block is messy. I would move initialization of `FileSystemPath`s to outside variables to keep this block simpler